pandas/core/generic.py

```diff
@@ -6720,6 +6722,28 @@ def interpolate(
                 "column to a numeric dtype."
             )
 
+        # Set `limit_direction` depending on `method`
```

Comment:
is there a reason this logic does not simply belong in interpolate_2d?

Reply:
or call something like clean_fill_method

Reply:
`limit_direction` has no effect in `interpolate_2d`.

My idea was to detect conflicting user input, e.g. `pad` with `limit_direction='backward'` as early as possible in the chain of calls to the main functions that are involved.

But I can move the logic to a function like `clean_fill_method`, e.g. called `set_limit_direction`. Should this function belong to generic.py or missing.py?

Reply:
interpolate_2d is also called from fillna. only NDFrame.interpolate allows limit_direction so does make sense to validate here.

clean_fill_method checks method. probably want to avoid passing more parameters around.

Reply:
pls make a method similr to clean_fill_method then, e.g this logic should living pandas/core/missing.py

Reply:
not sure what to do here. after merge with master, there is similar code at this location, but it is not mine...